### PR TITLE
[script.cinemavision] 1.1.1

### PR DESCRIPTION
**Pull request by ruuk**
### Description
- change:  Enhanced user interface
- add:     Ability to change interface theme
- add:     TMDB trailer scraper
- fix:     Compatibility fixes for Kodi v18
- add:     Trailers: Content scraper will now read genre and rating from .nfo files

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] 1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.cinemavision/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (enen92) [inline script.cinemavision/.vscode/settings.json]: please remove from the submission/ add .vscode to .gitignore

File: script.cinemavision/lib/main.py [hunk 2]
```diff
@@ -333,67 +354,115 @@ def onFirstInit(self):
...
-                        self.doMenu()
 
         except Exception:
             kodiutil.ERROR()
 
         kodigui.BaseWindow.onAction(self, action)
 
+    def onFocus(self, controlID):
+        if controlID == self.MENU_ADDON_SETTINGS_BUTTON_ID:
+            kodiutil.setGlobalProperty('option.hint', '[B]Preferences[/B]: Customize CinemaVision to your needs')
```
Comment (enen92) [inline script.cinemavision/lib/main.py]: this and the below strings need to be localized

File: script.cinemavision/lib/main.py [hunk 3]
```diff
@@ -409,8 +478,16 @@ def handleClose(self):
 
         if yes:
             return False
-
-        self.save(as_new=True)
+        else:
+            yes = xbmcgui.Dialog().yesno(
+                'Options',
+                'Would you like to save and exit or abort',
```
Comment (enen92) [inline script.cinemavision/lib/main.py]: same

File: script.cinemavision/lib/main.py [hunk 4]
```diff
@@ -448,40 +526,43 @@ def fillOptions(self):
...
+            item.setProperty('thumb.focus', '{0}small/script.cinemavision-{1}_Selected.png'.format(cvutil.THEME['theme.path'], i[2]))
+            item.setProperty('thumb.fill', '{0}small/script.cinemavision-{1}_Fill.png'.format(cvutil.THEME['theme.path'], i[2]))
             self.addItemControl.addItem(item)
 
-        item = kodigui.ManagedListItem(T(32531, 'Edit'), T(32531, 'Edit'), thumbnailImage='small/script.cinemavision-edit.png', data_source='edit')
-        item.setProperty('alt.thumb', 'small/script.cinemavision-edit.png')
-        item.setProperty('thumb.focus', 'small/script.cinemavision-A_selected.png')
+        basePath = cvutil.THEME['theme.path'] + 'options/script.cinemavision-'
+
+        item = kodigui.ManagedListItem(T(32531, 'Edit'), T(32531, 'Edit'), thumbnailImage=basePath + 'ModuleEdit.png', data_source='edit')
```
Comment (enen92) [inline script.cinemavision/lib/main.py]: if thumbnailImage is later used as the ListItem kwarg please drop and use only setArt

File: script.cinemavision/lib/main.py [hunk 5]
```diff
@@ -886,24 +1044,21 @@ def save(self, as_new=False, export=False):
...
             path = cinemavision.util.pathJoin(contentPath, 'Sequences')
 
-        name = self.name
+        pathName = self.sequenceData.pathName
 
-        if not name or as_new or export:
-            name = xbmcgui.Dialog().input(T(32554, 'Enter name for file'), name)
-            if not name:
+        if not pathName or as_new or export:
+            pathName = xbmcgui.Dialog().input(T(32554, 'Enter name for file'), pathName)
```
Comment (enen92) [inline script.cinemavision/lib/main.py]: localize if possible

Comment (ruuk) [reply script.cinemavision/lib/main.py]: This one already is, unless you mean `pathName` which isn't localizable.

File: script.cinemavision/lib/main.py (hunk 4 above)
Comment (ruuk) [reply script.cinemavision/lib/main.py]: Done in the `kodigui` lib, so this these and everywhere else will ultimately use setArt.

**Comment by ruuk** [comment]
@enen92: Thanks for the tips and catching the missed localization. Changes made.

**Comment by ruuk** [comment]
> Appart from the comments I'm wondering if the print method calls can be changed to xbmc.log wherever possible

All of the print statements left are in libs and not part of code that will be run in the addon's context.

File: script.cinemavision/lib/main.py (hunk 5 above)
Comment (enen92) [reply script.cinemavision/lib/main.py]: I mean "enter name for file"

Comment (ruuk) [reply script.cinemavision/lib/main.py]: That's why it's wrapped in `T`: `T(32554, 'Enter name for file')`. The `T` function localizes, and the text is just there for reference. It isn't actually used in the function.

**Comment by ruuk** [comment]
> Can't the modules in `included-packages` be moved to script.modules?

Theoretically, but I don't have currently have the time to create and maintain those as modules.

**Comment by enen92** [comment]
Do they include any patch or is just upstream code?

**Comment by ruuk** [comment]
> Do they include any patch or is just upstream code?

I don't recall if I had to modify anything to work with Kodi (until the stuff in this PR), it's been a few years :smile:.

**Comment by ruuk** [comment]
@enen92: Is there anything that still needs to be changed on this?

**Comment by enen92** [comment]
It's fine. I don't like the modules being bundled but if you don't have time at the moment to fix it I'd say to keep it like it is. This will die out with Matrix and should be fixed by that time

**Comment by ruuk** [comment]
> It's fine. I don't like the modules being bundled but if you don't have time at the moment to fix it I'd say to keep it like it is. This will die out with Matrix and should be fixed by that time

@enen92: Happy to merge this myself, but this is still marked as changes requested.

**Pull request merged**